File: downloadmanagerdemo/src/main/java/com/wys/downloadmanagerdemo/MainActivity.java
package com.wys.downloadmanagerdemo;

import android.Manifest;
import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.ActivityCompat;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.RemoteViews;

import com.wys.downloadmanagerdemo.download.DownloadService;

import static android.Manifest.permission.READ_EXTERNAL_STORAGE;


public class MainActivity extends AppCompatActivity {
    private String notificationChannelId = "download";
    RemoteViews         remoteViews;
    NotificationManager notificationManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        Log.d("MainActivity:pack", getPackageName());
        remoteViews = new RemoteViews(getPackageName(), R.layout.notification_download);
        ActivityCompat.requestPermissions(this, new String[]{READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE}, 0);
    }

    public void download(View view) {
        int i = 0;
        while (i < 100) {
            Notification notification= createNotification(i);
            notificationManager.notify(2000, notification);
            i++;
        }

        //        DownloadService.startDownloadTask(this, 0, "test1");
    }

    public void downloadvideo(View view) {
        DownloadService.startDownloadTask(this, 0, "test2");
    }

    public void downloadapk(View view) {
        DownloadService.startDownloadTask(this, 0, "test3");
    }


    private Notification createNotification(int progress) {
        Notification.Builder builder;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(notificationChannelId, "jjjjj", NotificationManager.IMPORTANCE_HIGH);
            notificationManager.createNotificationChannel(channel);
            builder = new Notification.Builder(this, notificationChannelId);
        } else {
            builder = new Notification.Builder(this);
        }

        RemoteViews remoteViews = new RemoteViews(getPackageName(), R.layout.notification_download);
        builder.setSmallIcon(R.drawable.ic_launcher);
        builder.setContentText("content");
        builder.setContentTitle("TITLE");
        builder.setContent(remoteViews);
        builder.setProgress(100, progress, false);
        //        builder.setLargeIcon(BitmapFactory.decodeResource(getResources(), R.mipmap.ic_launcher));
        //        builder.setContent(remoteViews);
        return builder.build();
    }
}

File: notificaitontest/src/main/java/com/wys/notificaitontest/MainActivity.kt
package com.wys.notificaitontest

import android.app.Notification
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        sendNotification.setOnClickListener {
            sendNotify()
        }
    }

    private fun sendNotify() {
        val intent = Intent(this, MainActivity::class.java)
        var pi = PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT)

        val notificationManagerCompat = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        val notification = Notification.Builder(this, "sss")
        notification.setSmallIcon(R.mipmap.ic_launcher).setTicker("接收到一条新的消息").setContentText("测测星座").setContentTitle("title").setAutoCancel(true).setChannelId("sss").setContentIntent(pi)
        notificationManagerCompat.notify(1, notification.build())
    }

}
